src/track/serato/markers2.cpp

```diff
@@ -726,23 +862,34 @@ bool SeratoMarkers2::isBpmLocked() const {
 }
 
 void SeratoMarkers2::setBpmLocked(bool bpmLocked) {
-    for (const auto& pEntry : qAsConst(m_entries)) {
-        VERIFY_OR_DEBUG_ASSERT(pEntry) {
-            continue;
-        }
+    QList<SeratoMarkers2EntryPointer> newEntries;
 
-        if (pEntry->typeId() != SeratoMarkers2Entry::TypeId::BpmLock) {
-            continue;
-        }
-
-        auto pBpmLockEntry = std::static_pointer_cast<SeratoMarkers2BpmLockEntry>(pEntry);
-        pBpmLockEntry->setLocked(bpmLocked);
-        DEBUG_ASSERT(countEntriesByType(SeratoMarkers2Entry::TypeId::BpmLock) == 1);
-        return;
+    // Append COLOR entry
+    const SeratoMarkers2EntryPointer pColorEntry =
```

Comment:
The question is how is the order related to the hotcue numbers. 

I am afraid this loop will mass up the cue numbers if something is wrong ordered.
Maybe you should creates a test file that is intentionally wrong ordered and check how and if Serato recorders the entries.

My guess is they assign the numbers consecutively without reordering by type.

Reply:
For markers, the number is part of the entry (in contrast to the markers tag). But I can check what happens if the order is wrong.

Reply:
Also, the order should be kept in this case. And when setting cue points, the order is correct thanks to `QMap<int, Entry>::values()`.